### PR TITLE
Create an ATLAS service to get cached project

### PR DESCRIPTION
**Pull request by rldhont**
The Filter is used to change request parameters from WMS GetCapabilitiesAtlas and GetPrintAtlas to ATLAS GetCapabilities and GetPrint.

### CODE REVIEW

**Comment by dmarteau** [comment]
Ha ! using a service: interesting workaround !

**Comment by Gustry** [comment]
Can you update the readme.md with this new API?

File: atlasprintServer/atlasprintService.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        self.msg  = msg
+        self.code = code
+        QgsMessageLog.logMessage("Atlas print request error %s: %s" % (code,msg),"atlasprint",Qgis.Critical)
+
+    def formatResponse(self, response: QgsServerResponse) -> None:
+        """ Format error response
+        """
+        body = { 'status': 'fail', 'message': self.msg }
+        response.clear()
+        write_json_response(body, response, self.code)
```
Comment (Gustry) [inline atlasprintServer/atlasprintService.py]: With @dmarteau we were saying that if the code is 500 (5XX), this should stay on the server side and not added in the body. We can fix that later.

File: atlasprintServer/atlasprintFilter.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            return
+
+        # Check request to change atlas one
+        if 'REQUEST' not in params or params['REQUEST'].lower() not in ['getprintatlas', 'getcapabilitiesatlas']:
+            return
+
+        request = params['REQUEST'].lower()
+
+        handler.setParameter('SERVICE', 'ATLAS')
+        handler.setParameter('SERVICE', '1.0.0')
```
Comment (Gustry) [inline atlasprintServer/atlasprintFilter.py]:
```suggestion
        handler.setParameter('VERSION', '1.0.0')
```
?

File: atlasprintServer/atlasprintService.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+from pathlib import Path
+
+from qgis.server import (QgsService,
+                         QgsServerRequest,
+                         QgsServerResponse)
+
+from .core import print_atlas, AtlasPrintException
+
+
+def write_json_response( data: Dict[str,str], response: QgsServerResponse, code: int=200) -> None:
```
Comment (dmarteau) [inline atlasprintServer/atlasprintService.py]: This raise an exception: 
```
>   def write_json_response( data: Dict[str,str], response: QgsServerResponse, code: int=200) -> None:
        """ Write data as json response
E       NameError: name 'Dict' is not defined
```

If your are using type annotations, you need:
```
from typing import Dict
```

**Comment by rldhont** [comment]
@dmarteau tests passed

**Comment by rldhont** [comment]
@Gustry I have updated the README.md. Can you review it ?

**Pull request merged**